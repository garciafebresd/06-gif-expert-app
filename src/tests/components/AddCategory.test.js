import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Testing AddCategory component', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
    });

    test('should render AddCategory in snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should change the input text', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';

        input.simulate('change', {target: {value}});

        const p = wrapper.find('p').text().trim();

        expect( p ).toBe(value);
    });
    
    test('should not post information on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();
    });
    

    test('should call setCategories and clean the input text', () => {

        // simulate input chance
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {target: {value}});

        // simulate form submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        // setCategories must be called
        // expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        // Validando que lo que se llamo sea una funci√≥n
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );

        // the value must be empty string
        const p = wrapper.find('input').prop('value');
        expect( p ).toBe('');
    });
    
});


