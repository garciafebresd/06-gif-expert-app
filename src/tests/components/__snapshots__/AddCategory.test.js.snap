// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory component should render AddCategory in snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
